### ISSUE TITLE
[FEATURE] add option to forbid arbitrary HTTP status codes

### ISSUE THREAD

**Issue by niya3** [schemathesis/schemathesis#1385]
**Is your feature request related to a problem? Please describe.**
Linked with https://github.com/schemathesis/schemathesis/issues/826

I wan't to clearly know if i have issue with the authorisation header provided to schemathesis. But if `401` status code is declared in schema among server responses, then schemathesis considers that check was ok.

**Describe the solution you'd like**
It would be handy to pass a (comma-separated?) list of forbidden HTTP status codes to `schemathesis run` and mark check with such codes as failed if authorization header was provided.

**Describe alternatives you've considered**
Now i can _post factum_ collect VCR cassete and parse it with `yq`, e.g. via `yq e '.http_interactions[] | select( .status != "FAILURE") | select( .response.status.code == 401) | .id + " " + .response.status.code'`

**Comment by serieznyi** [comment]
I have same problem with Authorization header and my solution - define own `check`.

```python

SUCCESSFUL_STATUS_CODES = [200, 201]

@schemathesis.register_check
def successful_response_code(response, case):
    if not response.status_code in SUCCESSFUL_STATUS_CODES:
        raise AssertionError("Not successful response code")

```

Or use `check`, which throw exception if `case` have defined authorization header and `response` status code is 401


```python
@schemathesis.register_check
def authorization_is_work(response, case):
    if response.status_code == 401 and case.headers["Authorization"]:
        raise AssertionError("Authorization problem")
```

![image](https://user-images.githubusercontent.com/2323142/153225743-a0e0cbfe-d059-425a-9414-333b93a4e32b.png)

**Comment by Stranger6667** [comment]
Yeah, a similar issue was also mentioned in [this comment](https://github.com/schemathesis/schemathesis/issues/1147#issuecomment-979980389) in #1147.

> It would be handy to pass a (comma-separated?) list of forbidden HTTP status codes to schemathesis run and mark check with such codes as failed if authorization header was provided.

I think that this is something that could be solved by custom checks, exactly as @serieznyi proposed. Also, I think they might be built-in into Schemathesis because the issue is quite common.

On the other hand, I think it could be better to stop all tests if auth is supplied, but it doesn't work. To avoid wasting time on tests that are expected to pass the auth (as it is explicitly supplied) - which is clearly not something that the user wants (if the user wants to run tests without auth, then the CLI option should not be there).

What do you think?

**Comment by niya3** [comment]
@Stranger6667 yep, i've already added custom check and it's just enough solve my problems right now. In my opinion the "fail fast" option is clearly the preferable one. Feel free to close this issue.

**Comment by Stranger6667** [comment]
At the moment there is a warning for this matter, but I think it could be nice to control whether it is a warning or an immediate failure.

**Comment by Stranger6667** [comment]
Right now, the `ignored_auth` check should cover this use case. Additionally, there is a warning that will highlight the issue (it is configurable too)
